graspy/cluster/base.py

```diff
@@ -56,10 +56,17 @@ def predict(self, X, y=None):
         ari : float
             Adjusted Rand index. Only returned if y is given.
         """
-        return labels, ari
+        # Check if fit is already called
+        check_is_fitted(self, ['model_'], all_or_any=all)
+        labels = self.model_.predict(X)
 
-    @abstractmethod
-    def fit_predict(self, X, y=None):
+        if y is None:
+            return labels
+        else:
+            ari = adjusted_rand_score(y, labels)
+            return labels, ari
+
+    def fit_predict(self, X, y=None): # pragma: no cover
```

Comment:
why the no cover?

Reply:
Its for coverage. There is no way to instantiate an abstract class, so there is no way to test these methods directly. The comment just makes coverage skip these lines.

The actual unit tests for GaussianCluster and KMeansClusters runs these methods so its already covered there.